lib/youtube-dl.js

```diff
@@ -163,25 +163,37 @@ function call(video, args1, args2, options, callback) {
  * @param {Array.<String>} data
  */
 function filterData(data) {
-  var format = data[data.length - 1].split(' - ');
+  var info = JSON.parse(data);
+  var format = info.format.split(' - ');
+  function formatDuration(duration) {
+    var parts = [];
+    parts.push(duration % 60);
+    var minutes = Math.floor(duration / 60);
+    if (minutes > 0) {
+      parts.push(minutes % 60)
+      var hours = Math.floor(minutes / 60);
+      if (hours > 0) {
+          parts.push(hours);
+      }
+    }
+    return parts.reverse().join(':');
+  }
 
   return {
-    title       : data[0],
-    id          : data[1],
-    url         : data[2],
-    thumbnail   : data[3],
-    description : data.slice(4, data.length - 3).join('\n'),
-    filename    : data[data.length - 3],
-    duration    : data[data.length - 2],
-    itag        : parseInt(format[0], 10),
+    title       : info.title,
+    id          : info.id,
+    url         : info.url,
+    thumbnail   : info.thumbnail,
+    description : info.description,
+    filename    : info._filename,
+    duration    : formatDuration(info.duration),
```

Comment:
I would just set it to the duration in seconds, but that would break backwards compatibility. If you want me to update it, just say it.